drivers/wifi/esp_at/esp.h

```diff
@@ -107,10 +107,10 @@ extern "C" {
 #define ESP_MODE_AP		2
 #define ESP_MODE_STA_AP		3
 
-#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
-#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
-#else
+#if defined(WIFI_ESP_AT_VERSION_1_7) || defined(WIFI_ESP_AT_VERSION_2_0)
```

Comment:
This is incorrect. `AT+CWMODE=<mode>,[autoconnect]` appears to be be the default moving forward:
[v2.1.0 docs](https://docs.espressif.com/projects/esp-at/en/release-v2.1.0.0_esp32/AT_Command_Set/Wi-Fi_AT_Commands.html#at-cwmodesets-the-wi-fi-mode-station-softap-station-softap)
[v2.2.0 docs](https://docs.espressif.com/projects/esp-at/en/release-v2.2.0.0_esp32/AT_Command_Set/Wi-Fi_AT_Commands.html#set-command)
[v2.3.0 docs](https://espressif-docs.readthedocs-hosted.com/projects/esp-at/en/release-v2.3.0.0_esp32c3/AT_Command_Set/Wi-Fi_AT_Commands.html#set-command)
But this is explicitly opting into it for the old versions only.

Reply:
I agree that my original change was technically incorrect because I was testing it against AT version 2.1.0.0 while using `CONFIG_WIFI_ESP_AT_VERSION_2_0` (as higher versions did not exist)

Reply:
@JordanYates ,  it appears to be something outdated in firmware when comparing  the documentation, we have flashed a bunch of  these click shields here, and all of them returned `ERROR` after using the CWMODE with two arguments, it seems the V2.0 still uses the legacy CWMODE with single argument.

A addressed the missing `CONFIG_ ` part.

Reply:
@JordanYates Any feedback?

Reply:
> it seems the V2.0 still uses the legacy CWMODE with single argument

Yes, thats what I was implying. I had `CONFIG_WIFI_ESP_AT_VERSION_2_0` selected while the ESP firmware was running v2.1.0